applications/coreutils/tac.cpp

```diff
@@ -0,0 +1,170 @@
+#include <libsystem/cmdline/CMDLine.h>
+#include <libsystem/io/Path.h>
+#include <libsystem/io/Stream.h>
+#include <libutils/String.h>
+#include <libutils/StringBuilder.h>
+#include <libutils/Vector.h>
+
+#include <cstring>
+
+static bool before = false;
+static char *separator = nullptr;
+
+static const char *usages[] = {
+    "NAME...",
+    "[OPTION]... NAME...",
+    nullptr,
+};
+
+static CommandLineOption options[] = {
+    COMMANDLINE_OPT_HELP,
+    COMMANDLINE_OPT_STRING("separator", 's', separator,
+                           "Choose the separator[STRING] to be used instead of \\n",
+                           COMMANDLINE_NO_CALLBACK),
+    COMMANDLINE_OPT_BOOL("before", 'b', before,
+                         "Attach the separator before instead of after the string",
+                         COMMANDLINE_NO_CALLBACK),
+    COMMANDLINE_OPT_END};
+
+static CommandLine cmdline = CMDLINE(
+    usages,
+    options,
+    "Concatenate and print lines of file(s) in reverse.",
+    "If no filename provided read from input stream\nNote that command may not work as expected when \\0 is encountered");
+
+char *str_split(char *str, char *const sub_str)
+{
+    /*
+    Function to split string into parts with sub_str as delimiter
+    Similar to strtok in string.h
+    May not work as expected when the str contains '\0' in between
+    */
+    static char *start = NULL;
+
+    if(!start)
+    {
+        start = str;
+    }
+
+    if(!*start)
+    {
+        return NULL;
```

Comment:
You should return `nullptr` instead of `null`

Reply:
also at the end of the function I presume, where im currently returning `NULL`